apps/spruce/src/pages/projectSettings/tabs/GithubPermissionGroupsTab/transformers.ts

```diff
@@ -26,6 +32,10 @@ export const gqlToForm = ((data: ProjectSettingsQuery["projectSettings"]) => {
 
 export const formToGql = ((formState, isRepo, id) => ({
   ...(isRepo ? { repoId: id } : { projectId: id }),
+  githubAppAuth: {
+    appId: formState?.appCredentials?.githubAppAuth?.appId ?? 0,
+    privateKey: "",
+  },
```

Comment:
The tab needs it in order to figure out if the settings on the other tab defaults to the repo because they are coupled. We don't want to default one without the other. 
😞

Reply:
does this scenario happen if there's no Default to Repo button on the page?

Reply:
yes if they default the permission groups but no the mappings or vice versa, it becomes harder to keep in sync and easier to end up in a bad state

Reply:
Oh I think I misunderstood your question. The reason we need the repo data is because if app settings is defaulted to the repo, the backend automatically defaults this too, when this is defaulted to the repo, we want to show a message saying so instead of letting users still define groups.

Reply:
(And even if the project defaults to the repo, if the app is defined on the project, we still want to let them add groups.)